src/syntree.rs

```diff
@@ -384,19 +524,27 @@ fn parse_type<'t>(ctx: Context<'t>) -> ParseResult<'t, Type> {
             (ctx, Tuple(types))
         }
 
+        // List
         T::LeftBracket => {
+            // Only contains a single type.
```

Comment:
Maybe self explanatory from the `single` call to `parse_type`?

Reply:
I read these comments as "lists only contain a single type, so `let (..) = parse_type(..)?`".